core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala

```diff
@@ -0,0 +1,138 @@
+    setExternalClusterMode()
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1, host + ":" + port)
+    self
+  }
+
+  def setH2OCluster(hostPort: String): H2OConf = {
+    setExternalClusterMode()
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1, hostPort)
+    self
+  }
+
+  def h2oClusterHost = {
+    val value = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1)
+    if (value.isDefined) {
+      Option(value.get.split(":")(0))
+    } else {
+      None
+    }
+  }
+
+  def h2oClusterPort = {
+    val value = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1)
+    if (value.isDefined) {
+      Option(value.get.split(":")(1))
+    } else {
+      None
+    }
+  }
+
+
+  def setYARNQueue(queueName: String) = {
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_YARN_QUEUE._1, queueName)
+    self
+  }
+
+  def setH2ODriverPath(path: String): H2OConf = {
+    setExternalClusterMode()
+    logWarning("Using external cluster mode")
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_H2O_DRIVER._1, path)
+    self
+  }
+
+  def setNumOfExternalH2ONodes(numOfExternalH2ONodes: Int): H2OConf = {
+    sparkConf.set(PROP_EXTERNAL_H2O_NODES._1, numOfExternalH2ONodes.toString)
+    self
+  }
+
+  def setHDFSOutputDir(dir: String): H2OConf = {
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_HDFS_DIR._1, dir)
+    self
+  }
+
+
+  def setMapperXmx(mem: String): H2OConf = {
+    sparkConf.set(PROP_EXTERNAL_H2O_MEMORY._1, mem)
+    self
+  }
+
+
+  def externalConfString: String =
+    s"""Sparkling Water configuration:
+        |  backend cluster mode : ${backendClusterMode}
+        |  cloudName            : ${cloudName.getOrElse("Not set yet")}
+        |  cloud representative : ${h2oCluster.getOrElse("Not set, using cloud name only")}
+        |  clientBasePort       : ${clientBasePort}
+        |  h2oClientLog         : ${h2oClientLogLevel}
+        |  nthreads             : ${nthreads}""".stripMargin
+}
+
+object ExternalBackendConf {
+
+  val PROP_EXTERNAL_H2O_MEMORY = ("spark.ext.h2o.hadoop.memory", "6g")
+
+  val PROP_EXTERNAL_CLUSTER_HDFS_DIR = ("spark.ext.h2o.external.hdfs.dir", null.asInstanceOf[String])
+
+  /**
+    * If this option is set then h2o external cluster will be automatically started using the provided
+    * h2o driver on yarn
+    */
+  val PROP_EXTERNAL_CLUSTER_H2O_DRIVER = ("spark.ext.h2o.external.h2o.driver", null.asInstanceOf[String])
+
+  /** Yarn queue on which external cluster should be started */
+  val PROP_EXTERNAL_CLUSTER_YARN_QUEUE = ("spark.ext.h2o.external.yarn.queue", null.asInstanceOf[String])
+
+  /** ip:port of arbitrary h2o node to identify external h2o cluster */
+  val PROP_EXTERNAL_CLUSTER_REPRESENTATIVE = ("spark.ext.h2o.cloud.representative", null.asInstanceOf[String])
+
+  /** Number of nodes to wait for when connecting to external H2O cluster */
+  val PROP_EXTERNAL_H2O_NODES = ("spark.ext.h2o.external.cluster.num.h2o.nodes", null.asInstanceOf[String])
```

Comment:
Please, another commit to change this one. Plus it is not critical problem right now.